flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -129,13 +135,54 @@ private boolean observeFlinkJobStatus(FlinkDeployment flinkApp, Configuration ef
         }
     }
 
+    private boolean observeSavepointStatus(
+            FlinkDeployment flinkApp, Configuration effectiveConfig) {
+        SavepointInfo savepointInfo = flinkApp.getStatus().getJobStatus().getSavepointInfo();
+        if (savepointInfo.getTriggerId() == null) {
+            LOG.debug("Checkpointing not in progress");
+            return true;
+        }
+        SavepointFetchResult savepointFetchResult;
+        try {
+            savepointFetchResult = flinkService.fetchSavepointInfo(flinkApp, effectiveConfig);
+        } catch (Exception e) {
+            LOG.error("Exception while fetching savepoint info", e);
+            return false;
+        }
+
+        if (!savepointFetchResult.isTriggered()) {
+            String error = savepointFetchResult.getError();
+            if (error != null
+                    || SavepointUtils.gracePeriodEnded(operatorConfiguration, savepointInfo)) {
+                LOG.error("Savepoint trigger failure, setting reconciliation error");
+                savepointInfo.setTriggerId(null);
+                ReconciliationUtils.updateForReconciliationError(
+                        flinkApp, error != null ? error : "Savepoint status unknown");
+                return false;
+            }
+            LOG.info("Savepoint operation not running, waiting within grace period");
+        }
+        if (savepointFetchResult.getSavepoint() == null) {
+            LOG.info("Savepoint not completed yet");
```

Comment:
Does this need to set the trigger id of the savepoint info to null?

Reply:
no, this means the savepoint is not completed yet, we will need to observe again